dev-docs/RFCs/proposals/fixed-frame-coordinates-rfc.md

```diff
@@ -56,20 +61,31 @@ new Layer({
 
 Layers can currently specify coordinates in `COORDINATE_SYSTEM.LNGLAT`, or `COORDINATE_SYSTEM.METER_OFFSETS` (local East-North-Up coordinates from an origin).
 
-Implementation-wise, our layer shaders would use `@math.gl/geospatial` to generate a transform from fixed coordinates to a local ENU coordinate system, multiply that with any model matrix, and then treat the layer data `COORDINATE_SYSTEM.METER_OFFSETS`.
+Implementation-wise, rendering would still happen in "cartographic" our layer shaders would use `@math.gl/geospatial` to generate a transformation matrix from fixed coordinates to a local ENU coordinate system, multiply that with any model matrix, and then treat the layer data `COORDINATE_SYSTEM.METER_OFFSETS`.
 
 The local ENU coordinate system could have its origin in the current view state center.
 
-Other name ideas for the new coordinate system mode: `COORDINATE_SYSTEM.WGS84`, `COORDINATE_SYSTEM.WGS84_CARTESIAN`, `COORDINATE_SYSTEM.ECF`, ...
+Proposals for naming:
+
+- `COORDINATE_SYSTEM.WGS84`
+- `COORDINATE_SYSTEM.WGS84_CARTESIAN`
+- `COORDINATE_SYSTEM.CARTESIAN_WGS84`
+- `COORDINATE_SYSTEM.ECF`, ...
+
+Notes:
+- is that WGS84 defines both cartesian and cartographic coordinates.
+- Cesium uses WGS84 as it base coordinate system and just calls it `Cartesian`
+- Does deck.gl need both `CARTESIAN` and `WGS84_CARTESIAN`. The difference is the defined center point, ajd whether things work with MapViews and how cartographic coordinates map to Cartesian. Explore how we can unify as much as possible.
+- Aliasing `LNGLAT` to `WGS84_CARTOGRAPHIC` could make sense, though we may want to work through subtleties of how WGS84 is defined (Epochs, relation to WebMercator etc).
 
 
 ### Proposal: Fixed Frame (WGS84 Cartesian) View States
 
-Support geospatial viewports accept `{x, y, z}` objects.
+To enable specifying view states in cartesian, geospatial viewports need accept `{x, y, z}` objects.
 
 At a minimum, we could just convert a fixed frame x,y,z datum to a cartographic/geodesic datum and use it as a traditional geospatial view state.
 
-Consideration: There is currently some some support for a meter offset position in geospatial view states (possibly partially crippled). Worth looking at how that would align, and also see if things can be kept reasonably similar with other non-geospatial view states.
+Consideration: There is currently some some support for a meter offset `position` in geospatial view states (possibly partially broken). Worth looking at how that would align, and also see if things can be kept reasonably similar with other non-geospatial view states.
```

Comment:
`position` is not broken, though not very well documented.

I personally prefer not to overload existing props - it will open a can of worms. We can do one of:

1. add a view state field `cartesianCenter` to `WebMercatorViewport`
2. subclass the `WebMercatorViewport` to deal with WGS84 cartesian coordinates in the constructor/project/unproject methods.
3. export lnglat <-> WGS84 utilities (wrappers of `@math.gl/geospatial` utilities) and ask the application to do the conversion itself

I do want to point out that conceptually, viewport is shared by all layers, and it only concerns projection modes, not coordinate systems. For example, `WebMercatorViewport` does not provide "support" for `METER_OFFSETS` or `LNGLAT_OFFSETS`. `viewport.project` and `viewport.unproject` only deals with cartographic coordinates. Each layer can define their own coordinate system, therefore `layer.project` and `layer.unproject` can convert from and to a specific coordinate system. Since `viewState` is global and tied to the `MapView`, it is not really a thing to "support" a new coordinate system with the view state.

Reply:
This sounds pretty reasonable for the short term, where we just add light weight support for WGS84. Would you like to make a commit and articulate these ideas in the RFC, rather than me trying to imperfectly transcribe your ideas?

Reply:
More for longer term alignment, I would not expect that WebMercatorViewports would be primary viewports for WGS84, as the full 3D coordinate system allows proper FirstPersonViewports to navigate the world in a real 3D coordinate system with all the benefits that bring - in particular when we have non-flat basemaps. But that is just my initial point of view, let's see how it looks once we start digging into the details.